app/user.go

```diff
@@ -1711,7 +1711,7 @@ func (a *App) SearchUsersInTeam(teamId string, term string, options *model.UserS
 		if err != nil {
 			return nil, err
 		}
-		if len(listOfAllowedChannels) == 0 {
+		if listOfAllowedChannels != nil && len(listOfAllowedChannels) == 0 {
```

Comment:
The fact that this is possible after checking for errors seems un-idiomatic. Do you think `(a *App).GetViewUsersRestrictions` should be updated instead like so?

```diff
diff --git a/app/user.go b/app/user.go
index 918345f40..95d643e45 100644
--- a/app/user.go
+++ b/app/user.go
@@ -2104,11 +2104,11 @@ func (a *App) GetViewUsersRestrictions(userId string) (*model.ViewUsersRestricti
 
 func (a *App) GetViewUsersRestrictionsForTeam(userId string, teamId string) ([]string, *model.AppError) {
 	if a.HasPermissionTo(userId, model.PERMISSION_VIEW_MEMBERS) {
-		return nil, nil
+		return []string, nil
 	}
 
 	if a.HasPermissionToTeam(userId, teamId, model.PERMISSION_VIEW_MEMBERS) {
-		return nil, nil
+		return []string, nil
 	}
 
 	result := <-a.Srv.Store.Channel().GetMembersForUser(teamId, userId)
```

Reply:
The thing is that there are three possible cases as for how `GetViewUsersRestrictions` works when there are no errors. If the result is `nil`, that means that there are no restrictions. If it's `[]string`, that means that the user cannot see any channel, and if it's an array with values, those are the channels that the user can see. Making your proposed change would change the behavior of the function, as for cases where you have all permissions, you'd return empty slice, which means "permissions to see nothing".

The problem here was that I was assuming that if the lenght of the response was 0, that meant case 2, but golang treats `nil` responses for `[]string` as `[]string(nil)`, which behaves both as a `nil` value and as a listh with 0 elements. That's why the check needs to go first for `nil` and then for the length.

Reply:
I didn't realize that `nil` and empty slice were meaningfully distinct return values from `GetViewUsersRestrictions`. Without any documentation on that `App` method it's not obvious.

Reply:
You're right, I'll add it as documentation on the method to clarify. Thanks!!